jpo-geojsonconverter/src/main/java/us/dot/its/jpo/geojsonconverter/converter/bsm/BsmTopology.java

```diff
@@ -0,0 +1,82 @@
+import us.dot.its.jpo.geojsonconverter.pojos.geojson.bsm.DeserializedRawBsm;
+import us.dot.its.jpo.geojsonconverter.pojos.geojson.bsm.ProcessedBsm;
+import us.dot.its.jpo.geojsonconverter.serialization.JsonSerdes;
+import us.dot.its.jpo.geojsonconverter.validator.BsmJsonValidator;
+import us.dot.its.jpo.geojsonconverter.validator.JsonValidatorResult;
+
+/**
+ * Kafka Streams Topology builder for processing BSM messages from
+ * ODE BSM JSON -> BSM GeoJSON
+ */
+public class BsmTopology {
+
+    private static final Logger logger = LoggerFactory.getLogger(BsmTopology.class);
+
+    public static Topology build(String bsmOdeJsonTopic, String bsmProcessedJsonTopic, BsmJsonValidator bsmJsonValidator) {
+        StreamsBuilder builder = new StreamsBuilder();
+
+        // Stream for raw BSM messages
+        KStream<Void, Bytes> rawOdeBsmStream =
+            builder.stream(
+                bsmOdeJsonTopic,
+                Consumed.with(
+                    Serdes.Void(),  // Raw topic has no key
+                    Serdes.Bytes()  // Raw JSON bytes
+                )
+            );
+
+        // Validate the JSON and write validation errors to the log at warn level
+        // Passes the raw JSON along unchanged, even if there are validation errors.
+        KStream<Void, DeserializedRawBsm> validatedOdeBsmStream = 
+            rawOdeBsmStream.mapValues(
+                (Void key, Bytes value) -> {
+                    DeserializedRawBsm deserializedRawBsm = new DeserializedRawBsm();
+                    try {
+                        JsonValidatorResult validationResults = bsmJsonValidator.validate(value.get());
+                        deserializedRawBsm.setOdeBsmData(JsonSerdes.OdeBsm().deserializer().deserialize(bsmOdeJsonTopic, value.get()));
+                        deserializedRawBsm.setValidatorResults(validationResults);
+                        logger.debug(validationResults.describeResults());
+                    } catch (Exception e) {
+                        JsonValidatorResult validatorResult = new JsonValidatorResult();
+
+                        validatorResult.addException(e);
+                        deserializedRawBsm.setValidationFailure(true);
+                        deserializedRawBsm.setValidatorResults(validatorResult);
+                        deserializedRawBsm.setFailedMessage(e.getMessage());
+
+                        logger.error("Error in bsmValidation:", e);
+                    }
+                    return deserializedRawBsm;
+                }
+            );
+
+        // Convert ODE BSM to GeoJSON
+        KStream<String, ProcessedBsm<Point>> processedJsonBsmStream =
+            validatedOdeBsmStream.transform(
+                () -> new BsmProcessedJsonConverter()
+            );
+
+            processedJsonBsmStream.to(
+            // Push the joined GeoJSON stream back out to the BSM GeoJSON topic 
+            bsmProcessedJsonTopic, 
+            Produced.with(
+                Serdes.String(),
```

Comment:
The only comment I have is to wrap the key in an object if we want to add fields to it down the line. This way when the serdes is applied it'll change the key as follows:

"172.18.0.1"  -> { "originIp": "172.18.0.1" }

Also we may want to add more fields in the key such as: source (log file/rsu), sanitized, etc.

Reply:
I am pretty sure both the RSU IP and the log file name would be scrubbed with the PPM filtered BSM topic so this might not work at all with the filtered. I don't think MAP and SPaT would pass privacy protection either. We might just go with the design that you need to run the geojsonconverter on the non-filtered topics.

Reply:
I am turning this key into an object as you requested and supporting the log file name.

Reply:
Yeah, I think the PPM only processes BSMs & TIMs so the list of topics would only be for the Processed BSM topology. I think it may be a bit out of scope of this PR but figured i'd bring it up!

Reply:
I would suggest that the key object implement the `RsuIdKey` interface, and write it to the topic using the `RsuIdPartitioner`.  That would save clients a step of repartitioning vs having the bsms from the same rsu go to random partitions.

Reply:
Alternately.. if this needs to run in PPM mode with the RSU id's removed, you could use the BSM id (coreDta.id) in both the key and for partitioning.  So your key object could be something like:

{
  rsuId: "127.0.0.7", // May be null
  bsmId: "AEF0101"
}

You could create an interface for the above object, and a partitioner that partitions on both fields with reasonable behavior if the rsuId is missing.

Reply:
Addressed

Reply:
@iyourshaw I have added bsmId as you suggested. The object I implemented for the key does indeed implement the RsuIdKey interface.